### DISCUSSION TITLE
Dynamically getting the number of elements

### DISCUSSION THREAD

**Discussion by ikit** [cypress-io/cypress#18299] [Component Testing]
Hi, 

Is there a way to dynamically get the number of element with cypress ?
I see lot of solution on Internet ( like[ this one](https://stackoverflow.com/questions/55335788/in-cypress-io-how-can-i-count-a-number-of-elements-without-failing-the-test-if) or  [ this one ](https://stackoverflow.com/questions/50609636/how-to-count-dom-elements-in-cypress-assertion) ), but no one works for me.

The HMI I have to is plugged to nightly dump of a SAP database with lot of data. I can't create a "simple data set to run my tests" and hardcode test values (like in that test `cy.get("table > tbody").find("tr").should("have.length", 4);`). My tests have to be flexible and I have to compute dynamically some value.

By example, I would like to check if the delete button of a element in a list is working.
For this one I would like to first count the number of element in my list, and then after clicking of the delete button of the item, checking that the new lenght of the list is the former lenght - 1.

How do this kind of test ?

**Comment by bahmutov** [comment]
How about something like this https://glebbahmutov.com/cypress-examples/8.5.0/recipes/add-list-item.html

**Reply by wp520133** [reply]
我跟我有着同样的一个疑问，我解决了好几天，没有搞定呢
I have the same question with me. I have solved it for several days, but I haven't done it yet
